src/figuresClasses.ts

```diff
@@ -1,19 +1,104 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'rectangle';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+    ) {
+      throw new Error('All sides of a triangle mush be positive numbers');
+    }
+
+    if (
+      a + b <= c
+      || b + c <= a
+      || a + c <= b
+    ) {
+      throw new Error('The longest side of a triangle must be smaller'
+      + 'than a sum of two others');
```

Comment:
Don't split string with + . Will be hard to work or find it in the future
```suggestion
      throw new Error(
        'The longest side of a triangle must be smaller than a sum of two others'
      );
```